eth_account/_utils/signing.py

```diff
@@ -44,6 +58,175 @@ def sign_transaction_dict(eth_key, transaction_dict):
     return (v, r, s, encoded_transaction)
 
 
+#
+# EIP712 Functionalities
+#
+def dependencies(primaryType, types, found=None):
+    """
+    Recursively get all the dependencies of the primaryType
+    """
+    # This is done to avoid the by-reference call of python
+    found = found or []
+
+    if primaryType in found:
+        return found
+    if primaryType not in types:
+        return found
+
+    found.append(primaryType)
+    for field in types[primaryType]:
+        for dep in dependencies(field["type"], types, found):
+            if dep not in found:
+                found.push(dep)
+
+    return found
+
+
+def dict_to_type_name_converter(field):
+    """
+    Given a dictionary ``field`` of type {'name': NAME, 'type': TYPE},
+    this function converts it to ``TYPE NAME``
+    """
+    return field["type"] + " " + field["name"]
+
+
+def encodeType(primaryType, types):
+    # Getting the dependencies and sorting them alphabetically as per EIP712
+    deps = dependencies(primaryType, types)
+    deps_without_primary_type = list(filter(lambda x: x != primaryType, deps))
+    sorted_deps = [primaryType] + sorted(deps_without_primary_type)
+
+    result = ''.join(
```

Comment:
@kclowes I actually don't want to use another function here because, it would be small code so as to encapsulate in a function. Further the reason that I have used `list comprehension` was to speed up the process which would be gone if I were to use a function.
But nevertheless, please let me know if you had any other intention for putting it inside the function. Will change it.